x/claims/keeper/ibc_callbacks_test.go

```diff
@@ -597,107 +412,66 @@ func (suite *KeeperTestSuite) TestReceive() {
-			},
-		},
-		{
-			"invalid ACK",
-			func() {
-				err := suite.app.ClaimsKeeper.OnAcknowledgementPacket(suite.ctx, mockpacket, []byte{})
-				suite.Require().Error(err)
-			},
-		},
-		{
-			"non ics20 packet",
-			func() {
-				err := suite.app.ClaimsKeeper.OnAcknowledgementPacket(suite.ctx, mockpacket, ack.Acknowledgement())
-				suite.Require().Error(err)
+				// check that the record is not deleted
+				suite.Require().True(suite.app.ClaimsKeeper.HasClaimsRecord(suite.ctx, secpAddr))
 			},
 		},
 		{
-			"no-op: error Ack",
+			"case 4: no-op - sender different than recipient, no claims records",
 			func() {
-				err := sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data")
-				ack := transfertypes.NewErrorAcknowledgement(err)
-				err = suite.app.ClaimsKeeper.OnAcknowledgementPacket(suite.ctx, mockpacket, ack.Acknowledgement())
-				suite.Require().NoError(err)
+				transfer := transfertypes.NewFungibleTokenPacketData("aevmos", "100", senderStr, receiverStr)
+				bz := transfertypes.ModuleCdc.MustMarshalJSON(&transfer)
+				packet := channeltypes.NewPacket(bz, 1, transfertypes.PortID, "channel-0", transfertypes.PortID, types.DefaultAuthorizedChannels[0], timeoutHeight, 0)
+
+				suite.Require().False(suite.app.ClaimsKeeper.HasClaimsRecord(suite.ctx, sender))
+				suite.Require().False(suite.app.ClaimsKeeper.HasClaimsRecord(suite.ctx, receiver))
+
+				resAck := suite.app.ClaimsKeeper.OnRecvPacket(suite.ctx, packet, ack)
+				suite.Require().True(resAck.Success())
 			},
 		},
 		{
-			"error - claim IBC action with no escrowed funds",
+			"case 4: no-op - same sender with EVM channel, no claims record",
 			func() {
-				addr, err := sdk.AccAddressFromBech32("evmos1x2w87cvt5mqjncav4lxy8yfreynn273xn5335v")
-				suite.Require().NoError(err)
-
-				mockpacket.Data = transfertypes.ModuleCdc.MustMarshalJSON(
-					&transfertypes.FungibleTokenPacketData{
-						Sender:   "evmos1x2w87cvt5mqjncav4lxy8yfreynn273xn5335v",
-						Receiver: "cosmos1qql8ag4cluz6r4dz28p3w00dnc9w8ueulg2gmc",
-					},
-				)
+				transfer := transfertypes.NewFungibleTokenPacketData("aevmos", "100", secpAddrCosmos, secpAddrEvmos)
+				bz := transfertypes.ModuleCdc.MustMarshalJSON(&transfer)
+				packet := channeltypes.NewPacket(bz, 1, transfertypes.PortID, "channel-0", transfertypes.PortID, types.DefaultEVMChannels[0], timeoutHeight, 0)
 
-				cr := types.NewClaimsRecord(sdk.NewInt(200))
-				suite.app.ClaimsKeeper.SetClaimsRecord(suite.ctx, addr, cr)
-				err = suite.app.ClaimsKeeper.OnAcknowledgementPacket(suite.ctx, mockpacket, ack.Acknowledgement())
-				suite.Require().Error(err)
+				resAck := suite.app.ClaimsKeeper.OnRecvPacket(suite.ctx, packet, ack)
+				suite.Require().True(resAck.Success(), string(resAck.Acknowledgement()))
 			},
 		},
 	}
 	for _, tc := range testCases {
 		suite.Run(fmt.Sprintf("Case %s", tc.name), func() {
-			suite.SetupTest() // reset
+			suite.SetupClaimTest() // reset
```

Comment:
I don't understand the difference with `SetupTest`?

Reply:
@fedekunze SetupClaimTest() is a helper that was introduced before this PR and additionally funds the module account with escrowed coins. I look into refactoring this now